### PR TITLE
Support loading signing keys from a file

### PR DESCRIPTION
**Pull request by lann**
Not sure this completely solves #145 from a "good UX" perspective, but it does provide a workaround.

Also tightened up handling of sensitive encoded private keys a bit, making sure those get zeroized on drop as well.

### CODE REVIEW

File: crates/crypto/src/signing/private_key.rs [hunk 1]
```diff
@@ -17,6 +17,12 @@ pub enum PrivateKeyInner {
 }
 
 impl PrivateKey {
+    /// Read a private key from a file.
+    pub fn from_file(path: impl AsRef<Path>) -> Result<Self, PrivateKeyParseError> {
```
Comment (Kylebrown9) [inline crates/crypto/src/signing/private_key.rs:21]: Is there a reason we need to add a `from_file` here and adding IO to the error set?
It feels like we're mixing abstractions and you could just as easily `read_to_string(...).parse()` to keep them separate.

Comment (Kylebrown9) [reply crates/crypto/src/signing/private_key.rs:21]: My main objection is the `PrivateKeyParseError::IO` addition which isn't actually caused by parsing or reachable when calling `parse`.

Comment (lann) [reply crates/crypto/src/signing/private_key.rs:21]: I stuck it in here because of `Zeroizing`, but I could probably shuffle things around to avoid it.

Comment (fibonacci1729) [reply crates/crypto/src/signing/private_key.rs:21]: Just curious bould this just return an `anyhow::Result` so as to not pollute `PrivateKeyParseError`? I find `from_file` conventions to be rather useful for public items like this but fine either way.

Comment (lann) [reply crates/crypto/src/signing/private_key.rs:21]: Switched `FromStr` -> `TryFrom<String>` which lets it zeroize.

Comment (lann) [reply crates/crypto/src/signing/private_key.rs:21]: Well you can [see the diff here](https://github.com/bytecodealliance/registry/compare/ed730919cf3f77f95e9e583963916cd8f0cb07c7..8d68811780d8e1b8f639ee8d919c5e4f8b875a62), but github seems to be having trouble with consistency... :grimacing:

File: crates/crypto/src/signing/private_key.rs [hunk 2]
```diff
@@ -58,16 +57,20 @@ impl fmt::Display for PrivateKey {
...
-        let bytes = STANDARD.decode(parts[1])?;
+    fn try_from(key: String) -> Result<Self, PrivateKeyParseError> {
+        let key = Zeroizing::new(key);
+
+        let Some((algo, b64_data)) = key.split_once(':') else {
+            return Err(PrivateKeyParseError::MissingColon)
+        };
+
+        let algo = algo.parse::<SignatureAlgorithm>()?;
+        let bytes = STANDARD.decode(b64_data)?;
```
Comment (lann) [inline crates/crypto/src/signing/private_key.rs:73]: Extra colon rejection is handled by base64 decode.

File: crates/crypto/src/signing/private_key.rs (hunk 1 above)
Comment (lann) [reply crates/crypto/src/signing/private_key.rs]: Had to push an empty diff to convince GH things had actually changed :roll_eyes: .

**Review by fibonacci1729 — APPROVED** [review]

**Comment by lann** [comment]
Replaced `FromStr` and `Display` impls with explicit `decode` and `encode` methods, which lets us use zeroize more and makes it less likely to accidentally display the key somewhere.

**Pull request merged**